src/main/java/uk/gov/justice/hmpps/offenderevents/services/ReceivePrisonerReasonCalculator.java

```diff
@@ -20,13 +24,36 @@ public Reason calculateReasonForPrisoner(String offenderNumber) {
             return Reason.RECALL;
         }
 
-        return switch (prisonerDetails.legalStatus()) {
+        final Optional<Reason> maybeRecallStatusFromProbation = switch (prisonerDetails.legalStatus()) {
+            case OTHER, UNKNOWN, CONVICTED_UNSENTENCED, SENTENCED, INDETERMINATE_SENTENCE -> calculateReasonForPrisonerFromProbation(offenderNumber);
+            default -> Optional.empty();
+        };
+
+        return maybeRecallStatusFromProbation.orElseGet(() -> switch (prisonerDetails.legalStatus()) {
             case RECALL -> Reason.RECALL;
             case CIVIL_PRISONER, CONVICTED_UNSENTENCED, SENTENCED, INDETERMINATE_SENTENCE -> Reason.CONVICTED;
             case IMMIGRATION_DETAINEE -> Reason.IMMIGRATION_DETAINEE;
             case REMAND -> Reason.REMAND;
             case DEAD, OTHER, UNKNOWN -> Reason.UNKNOWN;
```

Comment:
Should we remove any of the legal statuses from this switch that appear in the first switch as they appear to be redundant?

Reply:
Nope - calculateReasonForPrisonerFromProbation can return an empty